docs/spec/overview.md

```diff
@@ -1,70 +1,127 @@
 
-# Resource Types
+- A **Subscription** describes the transformation of an event and optional
+  forwarding of a returned event.
+
+- A **Source** allows an incoming events from an external system to be
+  _Subscribable_. A _Subscription_ is used to connect these events to
+  subsequent processing steps.
+
+- A **Channel** provides event persistance and fanout of events from a
+  well-known input address to multiple outputs described by _Subscriptions_.
+
+<!-- This image is sourced from https://drive.google.com/open?id=10mmXzDb8S_4_ZG_hcBr7s4HPISyBqcqeJLTXLwkilRc -->
+
+![Resource Types Overview](images/resource-types-overview.svg)
+
+- **Provisioners** implement strategies for realizing backing resources for
+  different implementations of _Sources_ and _Channels_ currently active in the
+  eventing system.
+
+<!-- This image is sourced from https://drive.google.com/open?id=1o_0Xh5VjwpQ7Px08h_Q4qnaOdMjt4yCEPixRFwJQjh8 -->
 
-The primary resources in the Knative Eventing _Feeds_ API are EventSource,
-EventType and Feed:
+![Resource Types Provisioners](images/resource-types-provisioner.svg)
 
-* **EventSource**, an archetype of an event producer.
+With extendibility and compostability as a goal of Knative Eventing, the
+eventing API defines several resources that can be reduced down to a well
+understood contracts. These eventing resource interfaces may be fulfilled by
+other Kubernetes objects and then composed in the same way as the concreate
+objects. The interfaces are ([Sinkable](interfaces.md#sinkable),
+[Subscribable](interfaces.md#subscribable),
+[Channelable](interfaces.md#channelable),
+[Targetable](interfaces.md#targetable)). For more details, see
+[Interface Contracts](interfaces.md).
 
-* **EventType**, the schema for an event. 
+## Subscription
 
-* **Feed**, the association between the output of an event producer to the
-  input of an event consumer.
+**Subscriptions** describe a flow of events from one event producer or
+forwarder (typically, _Source_ or _Channel_) to the next (typically, a
+_Channel_) through transformations (such as a Knative Service which processes
+CloudEvents over HTTP). A _Subscription_ controller resolves the addresses of
+transformations (`call`) and destination storage (`result`) through the
+_Targetable_ and _Sinkable_ interface contracts, and writes the resolved
+addresses to the _Subscribable_ `from` resource. _Subscriptions_ do not need to
+specify both a transformation and a storage destination, but at least one must
+be provided.
 
-An example _Feeds_ setup: GitHub would be the _EventSource_, a Pull Request
-notification would be the _EventType_ and the _Feed_ describes the _org/repo_,
-credentials, the specific _EventType_ and the consumer of the _Feed_ events.
+All event delivery linkage from a **Subscription** is 1:1 – only a single
+`from`, `call`, and `result` may be provided.
 
+For more details, see [Kind: Subscription](spec.md#kind-subscription).
```

Comment:
Does this imply that only a linear pipeline of function calls can be supported (regardless of the level of abstraction of deployment units)? Even if I manually deploy a set of _Subscription_, I could not introduce any conditional logic or branching?

Reply:
That seems so to my eyes. I think the most important "building block" is missing in the spec: the spec is missing a *"standard contract for a service to publish events on a bus channel, regardless of the bus implementation"*.

"Callable" services are *not* necessarily functions and moreover they are not necessarily *pure functions* without side effects. I think a Service should be able to publish one, two or several events within the context of a invocation, provided that it has been given the permissions (e.g. via static resource configuration) to do so.

Reply:
I agree, that I would expect a Service to be able to publish a number of events. This would also be convenient for error handling and enabling a different component to respond to errors within a Service.

Reply:
This has come up a couple times now. I think the correct answer is another spec that outlines the runtime contract spec for how subscribables, targetables and sinkables communicate at a protocol level and then how that maps to transports we support.

Reply:
Thanks for calling out this ambiguity. The intent is that _Sinkable_ would be the interface for delivering an event (including to a function) which takes ownership of the event from a "chain of responsibility" pattern.

I think there is a second document we need that describes the data plane transport contracts. For now, our baseline is HTTP POST of CloudEvents, but a protocol-negotiation scheme would be very interesting once we've shipped sufficient primitives.